client/src/app/common/CustomRules/rules-utils.tsx

```diff
@@ -11,12 +11,13 @@ export const checkRuleFileType = (filename: string): RuleFileType => {
     return "XML";
   } else return null;
 };
-
+type ParsedYaml = any[] | {};
```

Comment:
Since we are accessing the `labels` property on the parsed YAML array elements, can we use a type here that contains that property instead of `any`? Maybe something like:

```ts
type ParsedYamlElement = { labels?: string[]; };
type ParsedYaml = ParsedYamlElement[] | {};
```